pkg/gcs/gcs.go

```diff
@@ -0,0 +1,72 @@
+package gcs
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+type GCS interface {
+	GetObject(ctx context.Context, path string) ([]byte, error)
+	GetIndexHtmlPath(ctx context.Context, qID string) (string, error)
+}
+
+type Client struct {
+	client     *storage.Client
+	bucketName string
+}
+
+func New(ctx context.Context, bucketName string) (*Client, error) {
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		client:     client,
+		bucketName: bucketName,
+	}, nil
+}
+
+func (c *Client) GetIndexHtmlPath(ctx context.Context, qID string) (string, error) {
+	objs := c.client.Bucket(c.bucketName).Objects(ctx, &storage.Query{Prefix: qID + "/"})
+	return findIndexPage(qID, objs)
+}
+
+func (c *Client) GetObject(ctx context.Context, path string) ([]byte, error) {
+	obj := c.client.Bucket(c.bucketName).Object(path)
+	reader, err := obj.NewReader(ctx)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	datab, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return datab, nil
+}
+
+func findIndexPage(qID string, objs *storage.ObjectIterator) (string, error) {
+	page := ""
+	for {
+		o, err := objs.Next()
+		if err == iterator.Done {
```

Comment:
What will happen if we got an error other than iterator.Done? should we log the error? 
Is it possible that an error can prevent the iterator from continuing?

Reply:
From doc:
>In addition, if Next returns an error other than iterator.Done, all subsequent calls will return the same error. To continue iteration, a new `ObjectIterator` must be created.

I have changed it now to log the error and return not found to the user, do you agree?

Reply:
Yes, as long as you return on error, we could avoid deadloop and it should be ok